### DISCUSSION TITLE
teller as password manager

### DISCUSSION THREAD

**Discussion by dvgitit** [twpayne/chezmoi#3822] [Ideas]
Would it make sense, to add "teller" as a password manager? In that case chezmoi would gain more flexibility in volatile environments, where not all password managers are available. Probably "teller" is a special case, because it dispatches as well to other secret managers.

**Comment by halostatue** [comment]
I don't know what `teller` is, so I can't say. It might be something that can be resolved through the [secret](https://www.chezmoi.io/reference/templates/secret-functions/secret/) template function.

**Reply by twpayne** [reply]
I think the OP is referring to https://github.com/tellerops/teller.

Indeed, it looks like the `secret` template function would work here. However, `teller` has very little documentation AFAICT and there doesn't seem to be a mention of a way to return a single secret.

**Reply by halostatue** [reply]
It looks like `teller` is sort of cloud-connected `.env` reader based on a very quick read of the README.

You would use it like this:

```console
$ teller run --reset --shell -- chezmoi apply
```

Inside of your templates, you would use `env("SECRET_VARIABLE_NAME")` instead of calling something like `secret`. You could probably do something similar with 1Password using `op run -- chezmoi apply`.
